src/transaction.js

```diff
@@ -142,6 +142,15 @@ export default class Transaction extends EventEmitter {
   acquireConnection(client, config, txid) {
     const configConnection = config && config.connection
     return Promise.try(() => configConnection || client.acquireConnection().completed)
+    .catch(TimeoutError, (error) => {
```

Comment:
This is nice addition, earlier I was able to fill my pool and stuck waiting new transaction forever...